### PR TITLE
Migrate to new version_utils and cpe2pkg modules

### PR DESCRIPTION
**Pull request by msrb**

### CODE REVIEW

**Comment by msrb** [comment]
@tisnik this is kind of a mess, sorry about that :) I wasn't sure how to properly split it into multiple PRs.

What is going on in this PR:
* migrating existing code to previously introduced `version_utils` and `cpe2pkg` modules
    * part of this is also moving the functionality of figuring out version ranges for given CVE to a separate module (`version_identifier.py`)
* adding tests to the affected modules, where missing (and `victims.py` is still pending, that module will require some refactoring)

:)

**Comment by msrb** [comment]
[test]

**Comment by msrb** [comment]
[test]

**Comment by msrb** [comment]
[test]

**Comment by msrb** [comment]
[test]

**Comment by msrb** [comment]
[test]

**Comment by msrb** [comment]
Aand we are green :)

File: tests/identifiers/test_naive.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    We expect to always find the correct package name in the result set.
+    """
+    pkgfile_path = Path(__file__).parent.parent / Path('data/java-pkgfile')
+    identifier = NaivePackageNameIdentifier(
+        maven_vertx_cve, 'java', pkgfile_path, cpe2pkg_tool
+    )
+    candidates = identifier.identify()
+    assert len(candidates) == 10
+    for candidate in candidates:
+        if candidate.package == 'io.vertx:vertx-core':
```
Comment (tisnik) [inline tests/identifiers/test_naive.py]: assert 'io.vertx:vertx-core' in candidates?

File: cvejob/selectors/basic.py [hunk 2]
```diff
@@ -1,286 +1,80 @@
...
-        candidate = None
-        safe_version_range = list()
-        affected_version_range = list()
+        if not self.candidates:
+            return None
 
-        cpe_entries = self._get_cpe_entries()
+        cve_versions = self.get_versions_from_cve(self.cve)
+        if not cve_versions:
+            return True
```
Comment (tisnik) [inline cvejob/selectors/basic.py]: is this the expected return value and type?

File: cvejob/versions/__init__.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+"""Identifiers of affected and unaffected versions."""
+
+from cvejob.versions.version_identifier import NVDVersions
+
+assert NVDVersions
```
Comment (tisnik) [inline cvejob/versions/__init__.py]: ;)

File: cvejob/selectors/basic.py (hunk 2 above)
Comment (msrb) [reply cvejob/selectors/basic.py]: Good catch. If there are no versions in CVE (rare), then there is nothing we can evaluate in this selector. So we should just return the candidate with highest confidence score. Let me fix this :+1:

**Comment by msrb** [comment]
Updated. Please check @tisnik :wink:

**Pull request merged**